src/main/java/org/libsdl/app/SDLActivity.kt

```diff
@@ -374,4 +375,13 @@ open class SDLActivity internal constructor (context: Context?) : RelativeLayout
         mSurface.holder?.removeCallback(this) // should only happen on SDL_Quit
         nativeSurface.release()
     }
+
+    private fun processPendingSDLEvents(invocations: Int = 0) {
+        if (invocations > 5) return
+
+        this.nativeProcessEventsAndRender()
+        Handler(Looper.getMainLooper()).postDelayed({
+            processPendingSDLEvents(invocations + 1)
+        }, 100)
+    }
```

Comment:
I don't know if this has any effect to be honest. I felt like it makes things slightly better but the crash still exists (without the changes in UIScreen.swift)

Reply:
Would rather not some arbitrary timeouts like this. This may fix some things but hide others. We want to fix these issues predictably and deterministically rather than randomly.

Reply:
btw, it was not the timeout which fixed the problem. Calling `this.nativeProcessEventsAndRender()` "a few times" was the fix (twice is probably good enough most of the times). I thought it's a good idea to wait a few ms inbetween to mimic the behaviour of the choreographer. Still it concerns me that tasks in deinit do not get called. I agree that our approach with the Dispatch.sync is the correct approach here, but I find it unexpected that tasks in deinit remain unexecuted @ephemer